apps/studio/src/components/referralHero/ReferralDashboard.tsx

```diff
@@ -0,0 +1,287 @@
+/* eslint-disable react-hooks/rules-of-hooks */
+import { FunctionComponent } from "react";
+import {
+  Box,
+  Divider,
+  IconButton,
+  InputAdornment,
+  Modal,
+  Stack,
+  TextField,
+  Typography,
+  useTheme,
+} from "@mui/material";
+import CloseIcon from "@mui/icons-material/Close";
+import ContentCopyIcon from "@mui/icons-material/ContentCopy";
+import { useFlags } from "launchdarkly-react-client-sdk";
+import WalletConnectionOverlay from "./WalletConnectionOverlay";
+import ReferralDataBox from "./ReferralDataBox";
+import {
+  getReferralHeroUserCampaignData,
+  useAppDispatch,
+  useAppSelector,
+} from "../../common";
+import {
+  selectUi,
+  setIsConnectWalletModalOpen,
+  setIsReferralDashboardModalOpen,
+  setToastMessage,
+} from "../../modules/ui";
+import {
+  emptyProfile,
+  selectSession,
+  useGetProfileQuery,
+} from "../../modules/session";
+
+interface ReferralDashboardProps {
+  readonly campaignUUID: string;
+}
+
+/**
+ * ReferralDashboard component displays a modal with referral stats and sharable link.
+ * If the user doesn't have a wallet connected, it shows a wallet connection overlay.
+ */
+const ReferralDashboard: FunctionComponent<ReferralDashboardProps> = ({
+  campaignUUID,
+}) => {
+  const { webStudioArtistReferralCampaign } = useFlags();
+  // Check if the referral campaign is enabled before rendering the wrapper component
+  if (!webStudioArtistReferralCampaign) return null;
+
+  const theme = useTheme();
+  const dispatch = useAppDispatch();
+  const { isLoggedIn } = useAppSelector(selectSession);
+  const { isReferralDashboardModalOpen } = useAppSelector(selectUi);
+
+  // Saved wallet address will be used for referral rewards
+  const { data: { walletAddress } = emptyProfile } = useGetProfileQuery(
+    undefined,
+    { skip: !isLoggedIn }
+  );
```

Comment:
Switched from `wallet` connected conditional to `walletAddress`  conditional so that once a user already has a wallet address saved in their user profile there is no need to show the connect wallet dashboard blocker. 

Not sure if we want a dialog or something to say that the reward will go to the attached wallet addressed... will bring up to product.